### PR TITLE
HP-729 study-details component created

### PR DESCRIPTION
**Pull request by mitka0260**
Please review my code test component creating

### CODE REVIEW

**Comment by mitka0260** [comment]
please review my first code in TWN

File: angular-fe/src/app/_core/models/interfaces/main.ts [hunk 1]
```diff
@@ -34,4 +38,20 @@ export interface QueryError {
...
+
+export interface Content {
+	value: string;
+}
+
+export interface StudyImages {
+	title: string;
+	alt: string;
+	derivative: ImageUrl;
+}
```
Comment (madispold) [inline angular-fe/src/app/_core/models/interfaces/main.ts:51]: This is a general interface usable globally, should remove study module reference from name.

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	templateUrl: './study-detail.component.html',
+	styleUrls: ['./study-detail.component.scss'],
+})
+export class StudyDetailComponent {
+
+	public loading = true;
+	public studyPath = `/uuringud/${this.route.snapshot.params.id}`;
+	public study$: Observable<MappedStudyPage> = this.api.studyDetailViewQuery(this.studyPath)
+		.pipe(
+			tap(() => this.loading = false, error => console.log('Error: ', error)),
```
Comment (madispold) [inline angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.ts:20]: No actual error handling takes place, should make a global interceptor to direct to 404 page if response route === null.

File: angular-fe/src/app/_services/SettingsService.ts [hunk 3]
```diff
@@ -1,95 +1,95 @@
...
-  ) {
-    this.url = environment.API_URL;
-    this.ehisUrl = environment.EHIS_URL;
-  }
+	constructor(
+		private http: HttpClient,
+	) {
+		this.url = environment.API_URL;
+		this.ehisUrl = environment.EHIS_URL;
+	}
```
Comment (madispold) [inline angular-fe/src/app/_services/SettingsService.ts:15]: Need to use similar indenting configuration in IDEs. Project has been developed using:
**Indent using tabs, tab size 2.**

File: drupal/web/modules/custom/htm_custom_querymap_form/Maps/raw/studyListViewQuery.graphql [hunk 4]
```diff
@@ -25,7 +25,7 @@ query studyListViewQuery($nid: [String],
   $highlightedStudy: [String],
   $highlightedStudyEnabled: Boolean
   ) {
-  nodeQuery(offset: $offset, limit: $limit, sort: {field: $sortField, direction: $sortDirection, language: $lang},
+  nodeQuery(offset: $offset, limit: $limit, sort: {field: $sortField, direc`tion: $sortDirection, language: $lang},
```
Comment (madispold) [inline drupal/web/modules/custom/htm_custom_querymap_form/Maps/raw/studyListViewQuery.graphql:28]: This is a typo -> should check commits file-by-file in the future to find these

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.scss [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+.study__detail__labels {
+	display: block;
+	margin-top: 1rem;
+	&--highlight {
+		position: absolute;
+		top: -1rem;
+		margin-top: 0;
+	}
+}
```
Comment (madispold) [inline angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.scss:19]: Nice BEM pattern, but this style has no use in the template, is redundant.

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+				<labels
+					*ngIf="study.fieldCustomBoolean"
+					class="study__list__item__labels"
+					type="orange"
+					[data]="[study.fieldStudyTag]">
+				</labels>
+
+				<p>
+					<strong>{{study.fieldIntroduction.substring(0, 300)}}</strong>
```
Comment (madispold) [inline angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html:21]: Should be handled by BE, this limit is not for angular to handle. Also, should never use function calls in templates :)

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		<block [titleBorder]="false" *ngIf="study$ | async as study">
+			<block-title>{{ 'study.label' | translate }}</block-title>
+			<block-content [noPaddingTop]="true">
+
+				<h1>{{study.title}}</h1>
+				<share [pageTitle]="study?.title"></share>
+
+				<labels
+					*ngIf="study.fieldCustomBoolean"
+					class="study__list__item__labels"
```
Comment (madispold) [inline angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html:15]: Misleading class name and this particular class serves no purpose (no styles reference it), every bit of extra code should be removed if possible

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html (hunk 6 above)
Comment (madispold) [reply angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html:21]: Also, strong tag somehow adds font-weight of 700, the designs require 500.

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+				<!--				TODO button -->
+				<button>Loe edasi - näita vähem</button>
+
+				<images
+					*ngIf="study.fieldAdditionalImages"
+					[image]="study?.fieldAdditionalImages ? study.fieldAdditionalImages : []"
+				></images>
+
+				<p style="font-weight: 500">{{ 'study.reference_title' | translate }}:</p>
```
Comment (madispold) [inline angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html:35]: Should never write inline css styles (if possible), include in .scss file for uniform styles and 1 source of truth.

**Review by madispold — CHANGES_REQUESTED** [review]
Back to you :)

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.ts (hunk 2 above)
Comment (madispold) [reply angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.ts:20]: I will make the interceptor myself actually, then we can review it together later

File: angular-fe/src/app/_core/models/interfaces/main.ts (hunk 1 above)
Comment (mitka0260) [reply angular-fe/src/app/_core/models/interfaces/main.ts]: corrected to ImageEntity

File: angular-fe/src/app/_services/SettingsService.ts (hunk 3 above)
Comment (mitka0260) [reply angular-fe/src/app/_services/SettingsService.ts:15]: file .editorconfig in the root of the project should regulate these configurations

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.scss (hunk 5 above)
Comment (mitka0260) [reply angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.scss]: deleted, and was added class "bold"

File: angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html (hunk 7 above)
Comment (mitka0260) [reply angular-fe/src/app/modules/study/containers/study-detail/study-detail.component.html]: extra code was removed

**Pull request merged**